src/menus/EditMenus.cpp

```diff
@@ -616,209 +670,93 @@ void OnPaste(const CommandContext &context)
       dst->ClearAndPaste(t0, t1, src, true, true, &warper);
    };
 
-   auto clipTrackRange = srcTracks->Any();
-   auto pC = clipTrackRange.begin();
-   size_t nnChannels=0, ncChannels=0;
-   while (*pN && *pC) {
-      auto n = *pN;
-      auto c = *pC;
-      if (n->GetSelected()) {
-         bAdvanceClipboard = true;
-         if (mismatchedClip)
-            c = mismatchedClip;
-         if (!c->SameKindAs(*n)) {
-            if (!mismatchedClip) {
-               lastClipBeforeMismatch = prevClip;
-               mismatchedClip = c;
-            }
-            bAdvanceClipboard = false;
-            c = lastClipBeforeMismatch;
-
-
-            // If the types still don't match...
-            while (c && !c->SameKindAs(*n)) {
-               prevClip = c;
-               c = * ++ pC;
-            }
-         }
+   // Find tracks to paste in
+   auto correspondence = FindCorrespondence(tracks, *srcTracks);
+   if (correspondence.empty()) {
+      if (tracks.SelectedLeaders().size() == 1)
+         AudacityMessageBox(XO(
+"There are not enough tracks after the selected track for the copied tracks.")
+      );
+      else
+         AudacityMessageBox(XO(
+            "There are not enough selected tracks for the copied tracks.")
+      );
```

Comment:
If we keep this message (and end up deferring #4734 until later), we probably should point out both the problem and the solution. So: 

`"The content you are trying to paste will span across more tracks than you currently have available. Add more tracks and try again."`

and

`"There are not enough tracks selected to accommodate your copied content. Select additional tracks and try again."`